### PR TITLE
Splits out ebook generation logic into a seperate module

### PR DESCRIPTION
**Pull request by Zomega**
In anticipation of supporting multiple output formats, and other operations (e.g. checking to see if a story has been updated, reading in terminal).

Two notable changes or flaws I may need to tweak further:

1) I changed the default text from Helvetica to FreeSans because I didn't have Helvetica on my system and it was causing problems. Not sure if this is the right choice though. Feel free to tell me to nix that change.

2) No longer will cache base.css if the user is using a cached session. If this is a problem, I could move the session making code into it's own module to share it, but I figure given that it's a small site and github if fairly reliable, it's a small price to pay.

### CODE REVIEW

**Comment by Zomega** [comment]
Examining. :-)

File: ebook/cover.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import textwrap
 
 
-def make_cover(title, author, width=600, height=800, fontname="Helvetica", fontsize=40, bgcolor=(120, 20, 20), textcolor=(255, 255, 255), wrapat=30):
+def make_cover(title, author, width=600, height=800, fontname="FreeSans", fontsize=40, bgcolor=(120, 20, 20), textcolor=(255, 255, 255), wrapat=30):
```
Comment (kemayo) [inline ebook/cover.py:7]: I'm going to come back to this.

**Comment by kemayo** [comment]
I'm generally okay with all this.

I'm going to come back and change the font stuff after I merge it -- *I* don't have FreeSans on my system, after all. Seems like the sane approach would be to pick one of the "safe" fonts and have it be a fallback to whatever the user tried to pass in.

There's some unrelated stuff with adding the `url` to `Section`s, but I don't mind that and can see the use.

**Comment by kemayo** [comment]
I made 619a013 for cover font fallbacks.

**Comment by kemayo** [comment]
...and 8ac1aa8 so people can set their own font and other cover options, if my fallback choices were terrible.

**Pull request merged**